packages/server-wallet/src/chain-service/__chain-test__/chain-service.test.ts

```diff
@@ -198,6 +210,29 @@ describe('fundChannel', () => {
 });
 
 describe('registerChannel', () => {
+  it('dipatches a channel finalized event if the channel has been finalized BEFORE registering', async () => {
```

Comment:
I only tested the `Finalized` case as the `In Progress` case can be fragile to test due to timing issues.  I'd really hate to introduce another blinking test.

Reply:
I think there ought to be an issue tracking that. While I recognize the complexity, it would be good to have a solid test suite providing 100% coverage (in the sense of 100% of the chain's state)

Barring that, we ought to have a really good understanding of _why_ it will work in the `In Progress` case.

Reply:
I've added a test in db07285 by creating a `TestChainService` that can expose the `finalizingChannels` property.

This allows us to ensure that an "In progress" channel gets added to that list.

Reply:
There is a way to specify the timestamp with `evm_mine` rpc (`await provider.send('evm_mine', [timestampInSeconds]);`. We can:
1. Set channel storage to finalize the channel far in the future.
1. Register the channel.
1. Mine a block with a timestamp that is further in the future.
1. Check that the `channelFinalized` callback has fired.

Reply:
Done in 2f6b4fd